src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2397,6 +2397,88 @@ def load_key_and_certificates_from_pkcs12(self, data, password):
 
         return (key, cert, additional_certificates)
 
+    def serialize_key_and_certificates_to_pkcs12(self, password, name, key,
+                                                 cert, cas, key_encryption,
+                                                 cert_encryption):
+        if password is not None:
+            utils._check_byteslike("password", password)
+        if name is not None:
+            utils._check_byteslike("name", name)
+        if key is not None and not isinstance(
+            key, (rsa.RSAPrivateKeyWithSerialization,
+                  dsa.DSAPrivateKeyWithSerialization,
+                  ec.EllipticCurvePrivateKeyWithSerialization)):
+            raise TypeError(
+                "Key must be RSA, DSA, or EllipticCurve private key."
+            )
+        if cert is not None and not isinstance(cert, x509.Certificate):
+            raise TypeError("cert must be a certificate")
+
+        if key is None:
+            nid_key = 0
```

Comment:
Do we trust OpenSSL's default?

Reply:
No:

```
These defaults are: 40 bit RC2 encryption for certificates, triple DES encryption for private keys, a key iteration count of PKCS12_DEFAULT_ITER (currently 2048) and a MAC iteration count of 1.

The default MAC iteration count is 1 in order to retain compatibility with old software which did not interpret MAC iteration counts. If such compatibility is not required then mac_iter should be set to PKCS12_DEFAULT_ITER.

```
https://www.openssl.org/docs/man1.1.1/man3/PKCS12_create.html

Compatibility is likely to be a real problem here, but we should at least do AES before we merge this.